stdlib/database/sql/src/main/java/org/ballerinalang/database/sql/statement/AbstractSQLStatement.java

```diff
@@ -404,28 +407,30 @@ protected void cleanupResources(Statement stmt, Connection conn, boolean connect
                 conn.close();
             }
         } catch (SQLException e) {
-            throw new BallerinaException("error in cleaning sql resources: " + e.getMessage(), e);
+            throw new Error(errorMessagePrefix + "error in cleaning sql resources: " + e.getMessage(), e);
```

Comment:
How about we let the SQL exception to be thrown from this method and handle it where this is called? 

We can't throw java Error instances.
Quoting java documentation.

> An Error is a subclass of Throwable that indicates serious problems that a reasonable application should not try to catch. Most such errors are abnormal conditions. The ThreadDeath error, though a "normal" condition, is also a subclass of Error because most applications should not try to catch it.

> A method is not required to declare in its throws clause any subclasses of Error that might be thrown during the execution of the method but not caught, since these errors are abnormal conditions that should never occur.

Reply:
IMO, Since this cleanupResources method is getting called within catch blocks and finally blocks we need to throw BallerinaException here to make it a panic scenario.

Reply:
I was having doubts whether we should let it panic or return it. May be thats the way to go. Since there is nothing user can do about that scenario.